### ISSUE TITLE
Cannot de-select PDF from file types for which Reader should be the default viewer

### ISSUE THREAD

**Issue by dysonsphere-startmail** [e-alfred/epubreader#27]
In Settings > Additional Settings I try to de-select Reader from opening PDFs.
I get a "saving..." message that hangs.
The setting never sticks.

Nextcloud: 20.0.4
EPUB/CBZ/PDF ebook reader: 1.4.5

**Comment by JazMTaz** [comment]
Same issue for me here also using Nextcloud 20.0.4, EPUB/CBZ/PDF ebook reader: 1.4.5

**Comment by m0yP** [comment]
Same here! I have several shared PDFs and when accessing them it loads the default view (which is fine) but also a pop-up window with the reader!

**Comment by szaimen** [comment]
Would love this, too :)
Actually, this is the only issue that is preventing me from installing the app currently

**Comment by rhein** [comment]
This is actually the same bug as issue #20

**Comment by Fardilha** [comment]
Hi.

Just want to add my "me too" on this problem.
Running EPUB/CBZ/PDF ebook reader 1.4.6 on NextCloud 20.0.9 and I can't de-select the PDF option.

Also it wouldn't be a bad idea to have a global option to prevent EPUB/CBZ/PDF ebook reader of "overtake" the PDF on the whole server and for the whole users.

Thanks
Pedro

**Comment by Kevin8675** [comment]
Still having this issue in 1.4.7 with NC 23. I am able to deselect it, but it will just say "Saving..." and when I reload, it is selected again.

**Comment by sysadminstory** [comment]
I digged a little bit for this issue : after selecting / deselecting a format in the settings page, there is an AJAX  POST request on :
https://url.of.the.nextcloud.instance/index.php/apps/epubreader/ajax/personal.php

The POST request redirect the broswer to :
https://cloud.phpkernel.fr/apps/dashboard/

I think the "index.php" should not in the initial POST request.

My assumption is that the issue is in this line :
https://github.com/e-alfred/epubreader/blob/e4e5143b3ef4147b51627df3d4c168a22fd6b367/js/settings.js#L12

According to https://docs.nextcloud.com/server/19/developer_manual/app/upgrade-guide.html OC.filePath should be replaced by generateFilePath from https://www.npmjs.com/package/@nextcloud/router

Here is the doc of this new function :
https://nextcloud.github.io/nextcloud-router/modules/_index_.html#generatefilepath

**Comment by manuelkamp** [comment]
bump (is any dev here around and reading any comments?)

**Comment by TheNomad11** [comment]
yeah, such a useful app, it should be part of the core actually, but I had to uninstall again because of this issue.

**Comment by sysadminstory** [comment]
I searched a little bit why this does not work anymore.

In the App metadata, the route to save the settings is not defined like in this documentation :
https://docs.nextcloud.com/server/latest/developer_manual/basics/routing.html

I don't know how the settings are actually displayed, because there is no metadata set in this app, like in :
https://docs.nextcloud.com/server/latest/developer_manual/basics/setting.html

The code to handle the settings change is not done in a "Controller", but in a simple file : ajax/personal.php

This app need a rework to work fully in recent version of Nextcloud.

I may give it a try, but it will be a big challenge for me :o

**Comment by sysadminstory** [comment]
My PR  #41 should fix this issue.

I'm not a Nextcloud app developper, so this PR needs to be reviewed before being production ready !
